test/lib/providers/metrics-provider.spec.js

```diff
@@ -1,36 +1,242 @@
+/* eslint-disable max-statements,no-magic-numbers,max-nested-callbacks */
+
 "use strict";
 
 var expect = require("chai").expect;
 var sinon = require("sinon");
 var _ = require("lodash");
 
+var AGGREGATE_TIME_LEVELS = require("../../../lib/constants.js").AGGREGATE_TIME_LEVELS;
+
 var utils = require("../../utils");
 var MetricsProvider = require("../../../lib/providers/metrics-provider");
 
 describe("MetricsProvider", function () {
-
   var sandbox;
   var testContainer;
   var metricsProvider;
 
-  before(function () {
+  var stubNow;
+  var mockStart = 1000000;
+  var mockNow = 1000000;
+  var mockTimeInterval = 2500;
+
+  var metricsRequiredToAchieveHighestAggregation =
+    +AGGREGATE_TIME_LEVELS.slice(-1)[0] / mockTimeInterval;
+
+  var mockMetrics = [];
+  var mockMetricCount;
+
+  before(function (done) {
     sandbox = sinon.sandbox.create();
+
+    // generate some fake metrics for processing
+    mockMetricCount =
+      Math.ceil(Math.random() * 500 + metricsRequiredToAchieveHighestAggregation);
+
+    for (var index = 0; index < mockMetricCount; index++) {
+      mockMetrics.push({
+        metricA: {
+          valueA: Math.random() * 100
+        },
+        metricB: {
+          valueA: Math.random() * 100,
+          valueB: Math.random() * 100
+        }
+      });
+    }
+
+    done();
```

Comment:
Sounds good. I've always done the done() as a consistency thing, but easily removed!